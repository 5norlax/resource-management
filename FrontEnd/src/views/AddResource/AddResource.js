import React, { Component } from 'react';
import axios from '../../api/index';
import { Redirect } from 'react-router-dom';
import { Container, Form, Button } from 'reactstrap';
import Header from '../../components/layout/Header/Header';
import AboutForm from '../../components/forms/AboutForm/AboutForm';
import SummaryForm from '../../components/forms/SummaryForm/SummaryForm';
import EducationForm from '../../components/forms/EducationForm/EducationForm';
import CertificationForm from '../../components/forms/CertificationForm/CertificationForm';
import ExperienceForm from '../../components/forms/ExperienceForm/ExperienceForm';
import ProjectForm from '../../components/forms/ProjectForm/ProjectForm';
import SkillForm from '../../components/forms/SkillForm/SkillForm';
import ResumeUpload from '../../components/forms/ResumeUpload/ResumeUpload';
import SectionHeader from '../../components/layout/SectionHeader/SectionHeader';
import DynamicSectionHeader from '../../components/layout/DynamicSectionHeader/DynamicSectionHeader';

class AddResource extends Component {
    state = {
        data: {
            "Id": "",
            "FirstName": "",
            "LastName": "",
            "Role": "",
            "Email": "",
            "Phone": "",
            "LinkedIn": "",
            "GitHub": "",
            "PersonalSite": "",
            "SummaryText": "",
            "Education": [],
            "Experience": [],
            "Project": [],
            "Certification": [],
            "Skills": []
        },
        submitted: false,
        resume: '',
        files: null
    };
    
    handleResume = (e) => {
        let resume = e.target.value;
        let files = e.target.files;
        this.setState({ resume, files });
    };

    handleCurrentPositionChange = (index, e) => {
        let currentPosition = this.state.currentPosition
        currentPosition[index] = !currentPosition[index]
        this.setState({ currentPosition })
    };

    handleChange = (e) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data[e.target.name] = e.target.value
        this.setState({ data })
    };

    handleEducationChange = (index, e) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Education[index][e.target.name] = e.target.value
        this.setState({ data })
    };

    handleCertificationChange = (index, e) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Certification[index][e.target.name] = e.target.value
        this.setState({ data })
    };

    handleExperienceChange = (index, e) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Experience[index][e.target.name] = e.target.value
        this.setState({ data })
    };

    handleProjectChange = (index, e) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Project[index][e.target.name] = e.target.value
        this.setState({ data })
    };

    handleSubmit = (e) => {
        e.preventDefault()
        const data = JSON.parse(JSON.stringify(this.state.data))
        axios.post(`api/resources`, { data })
            .then(res => {
                this.setState({ submitted: true })
                if (this.state.files) {
                    const formData = new FormData();
                    formData.append('resume', this.state.files[0]);
                    axios.post(`resources/${res.data.Id}/resume`, formData);
                }
            })
    };

    handleSkillsChange = (skills) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Skills = skills
        this.setState({ data })
    };

    handleAddEducation = () => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Education.push(
            {
                "School": "",
                "Location": "",
                "GradDate": "",
                "Degree": "",
                "Major": "",
                "Minor": ""
            }
        )
        this.setState({ data })
    };

    handleRemoveEducation = (index) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        if (data.Education.length === 1 || data.Education.length === index) data.Education.pop()
        else data.Education = [...data.Education.slice(0, index), ...data.Education.slice(index + 1)]
        this.setState({ data })
    };

    handleAddCertification = () => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Certification.push(
            {
                "CertName": "",
                "CertDate": "",
                "CertAssociation": ""
            }
        )
        this.setState({ data })
    };

    handleRemoveCertification = (index) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        if (data.Certification.length === 1 || data.Certification.length === index) data.Certification.pop()
        else data.Certification = [...data.Certification.slice(0, index), ...data.Certification.slice(index + 1)]
        this.setState({ data })
    };

    handleAddExperience = () => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Experience.push(
            {
                "JobTitle": "",
                "JobOrg": "",
                "JobStartDate": "",
                "JobEndDate": "",
                "JobInfo": ""
            }
        )
        this.setState({ data })
    };

    handleRemoveExperience = (index) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        if (data.Experience.length === 1 || data.Experience.length === index) data.Experience.pop()
        else data.Experience = [...data.Experience.slice(0, index), ...data.Experience.slice(index + 1)]
        this.setState({ data })
    };
    
    handleAddProject = () => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        data.Project.push(
            {
                "ProjName": "",
                "ProjDate": "",
                "ProjAssociation": "",
                "ProjInfo": ""
            }
        )
        this.setState({ data })
    };

    handleRemoveProject = (index) => {
        let data = JSON.parse(JSON.stringify(this.state.data))
        if (data.Project.length === 1 || data.Project.length === index) data.Project.pop()
        else data.Project = [...data.Project.slice(0, index), ...data.Project.slice(index + 1)]
        this.setState({ data })
    };

    render() {
        return this.state.submitted ? <Redirect to="/resources" /> : (
            <div>
                
                <Header name={'Add Resource Form'} />
                <Container className="ResourceForm">
                    <Form onSubmit={this.handleSubmit}>
                        <SectionHeader name="About" />
                        <AboutForm handleChange={this.handleChange} FirstName={this.state.data.FirstName} LastName={this.state.data.LastName} Role={this.state.data.Role}
                            Email={this.state.data.Email} Phone={this.state.data.Phone} LinkedIn={this.state.data.LinkedIn} 
                            GitHub={this.state.data.GitHub} PersonalSite={this.state.data.PersonalSite} />
                        <SectionHeader name="Resume" />
                        <ResumeUpload resume={this.state.resume} handleResume={this.handleResume} existingResume={this.state.existingResume}
                        id={this.state.data.Id} FirstName={this.state.data.FirstName} LastName={this.state.data.LastName} />
                        <SectionHeader name="Summary" />
                        <SummaryForm handleChange={this.handleChange} SummaryText={this.state.data.SummaryText} />
                        <DynamicSectionHeader name="Education" count={this.state.data.Education.length} addForm={this.handleAddEducation} />
                        {this.state.data.Education.map((element, index) =>
                            <EducationForm key={index} index={index} 
                            removeEducation={this.handleRemoveEducation} handleEducationChange={this.handleEducationChange}
                            School={element.School} Location={element.Location} Degree={element.Degree} 
                            Major={element.Major} Minor={element.Minor} GradDate={element.GradDate} />
                        )}
                        <DynamicSectionHeader name="Certifications" count={this.state.data.Certification.length} addForm={this.handleAddCertification} />
                        {this.state.data.Certification.map((element, index) =>
                            <CertificationForm key={index} index={index} 
                            removeCertification={this.handleRemoveCertification} handleCertificationChange={this.handleCertificationChange}
                            CertName={element.CertName} CertDate={element.CertDate} CertAssociation={element.CertAssociation} />
                        )}
                        <DynamicSectionHeader name="Experience" count={this.state.data.Experience.length} addForm={this.handleAddExperience} />
                        {this.state.data.Experience.map((element, index) =>
                            <ExperienceForm key={index} index={index} 
                            removeExperience={this.handleRemoveExperience} handleExperienceChange={this.handleExperienceChange}
                            JobTitle={element.JobTitle} JobOrg={element.JobOrg} JobStartDate={element.JobStartDate} 
                            JobEndDate={element.JobEndDate} JobInfo={element.JobInfo} />
                        )}
                        <DynamicSectionHeader name="Projects" count={this.state.data.Project.length} addForm={this.handleAddProject} />
                        {this.state.data.Project.map((element, index) =>
                            <ProjectForm key={index} index={index} 
                            removeProject={this.handleRemoveProject} handleProjectChange={this.handleProjectChange}
                            ProjName={element.ProjName} ProjDate={element.ProjDate} 
                            ProjAssociation={element.ProjAssociation} ProjInfo={element.ProjInfo} />
                        )}
                        <SectionHeader name="Skills" />
                        <SkillForm skills={this.state.data.Skills} handleSkillsChange={this.handleSkillsChange} />
                        <Button type="submit" className="blue-button col-md-12" color="primary">Submit Form</Button>
                    </Form>
                </Container>
            </div>
        );
    };
};

export default React.memo(AddResource);