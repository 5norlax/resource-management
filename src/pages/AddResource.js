import React, { Component } from 'react';
import { Container, Form, Button } from 'reactstrap';
import Header from '../components/Header';
import AboutForm from '../components/forms/AboutForm';
import SummaryForm from '../components/forms/SummaryForm';
import EducationForm from '../components/forms/EducationForm';
import CertificationForm from '../components/forms/CertificationForm';
import ExperienceForm from '../components/forms/ExperienceForm';
import ProjectForm from '../components/forms/ProjectForm';
import SkillForm from '../components/forms/SkillForm';
import SectionHeader from '../components/SectionHeader';
import DynamicSectionHeader from '../components/forms/DynamicSectionHeader';

class AddResource extends Component {
    state = {
        skills: ['React.js', 'Python'],
        EducationForms: [],
        CertificationForms: [],
        ExperienceForms: [],
        ProjectForms: []
    }

    handleSubmit = (e) => {
        e.preventDefault()
    }

    handleAddSkill = (skill) => {
        this.setState({
            skills: [...this.state.skills, skill]
        })
    }

    handleRemoveSkill = (skill) => {
        let skills = this.state.skills.filter(s => s !== skill)
        this.setState({
            skills: skills
        })
    }

    handleAddEducation = () => {
        let index = this.state.EducationForms.length
        this.setState({
            EducationForms: [...this.state.EducationForms, <EducationForm key={index} index={index} removeEducation={this.handleRemoveEducation} />]
        })
    }

    handleRemoveEducation = (index) => {
        let form = []
        if (this.state.EducationForms.length === 1 || this.state.EducationForms.length === index) {
            form = [...this.state.EducationForms]
            form.pop()
        }
        else form = [...this.state.EducationForms.slice(0, index), ...this.state.EducationForms.slice(index + 1)]
        this.setState({
            EducationForms: form
        })
    }

    handleAddCertification = () => {
        let index = this.state.CertificationForms.length
        this.setState({
            CertificationForms: [...this.state.CertificationForms, <CertificationForm key={index} index={index} removeCertification={this.handleRemoveCertification} />]
        })
    }

    handleRemoveCertification = (index) => {
        let form = []
        if (this.state.CertificationForms.length === 1 || this.state.CertificationForms.length === index) {
            form = [...this.state.EducationForms]
            form.pop()
        }
        else form = [...this.state.CertificationForms.slice(0, index), ...this.state.CertificationForms.slice(index + 1)]
        this.setState({
            CertificationForms: form
        })
    }

    handleAddExperience = () => {
        let index = this.state.ExperienceForms.length
        this.setState({
            ExperienceForms: [...this.state.ExperienceForms, <ExperienceForm key={index} index={index} removeExperience={this.handleRemoveExperience} />]
        })
    }

    handleRemoveExperience = (index) => {
        let form = []
        if (this.state.ExperienceForms.length === 1 || this.state.ExperienceForms.length === index) {
            form = [...this.state.EducationForms]
            form.pop()
        }
        else form = [...this.state.ExperienceForms.slice(0, index), ...this.state.ExperienceForms.slice(index + 1)]
        this.setState({
            ExperienceForms: form
        })
    }

    handleAddProject = () => {
        let index = this.state.ProjectForms.length
        this.setState({
            ProjectForms: [...this.state.ProjectForms, <ProjectForm key={index} index={index} removeProject={this.handleRemoveProject} />]
        })
    }

    handleRemoveProject = (index) => {
        let form = []
        if (this.state.ProjectForms.length === 1 || this.state.ProjectForms.length === index) {
            form = [...this.state.EducationForms]
            form.pop()
        }
        else form = [...this.state.ProjectForms.slice(0, index), ...this.state.ProjectForms.slice(index + 1)]
        this.setState({
            ProjectForms: form
        })
    }

    render() {
        return (
            <div>
                <Header name={'Resource Skills Form'} />
                <Container className="ResourceForm">
                    <Form onSubmit={this.handleSubmit}>
                        <SectionHeader name="About" />
                        <AboutForm />
                        <SectionHeader name="Summary" />
                        <SummaryForm />
                        <DynamicSectionHeader name="Education" count={this.state.EducationForms.length} addForm={this.handleAddEducation} />
                        {this.state.EducationForms}
                        <DynamicSectionHeader name="Certifications" count={this.state.CertificationForms.length} addForm={this.handleAddCertification} />
                        {this.state.CertificationForms}
                        <DynamicSectionHeader name="Experience" count={this.state.ExperienceForms.length} addForm={this.handleAddExperience} />
                        {this.state.ExperienceForms}
                        <DynamicSectionHeader name="Projects" count={this.state.ProjectForms.length} addForm={this.handleAddProject} />
                        {this.state.ProjectForms}
                        <SectionHeader name="Skills" />
                        <SkillForm skills={this.state.skills} addSkill={this.handleAddSkill} removeSkill={this.handleRemoveSkill} />
                        <Button type="submit" className="col-md-12" color="primary">Submit Form</Button>
                    </Form>
                </Container>
            </div>
        );
    }
}

export default AddResource;