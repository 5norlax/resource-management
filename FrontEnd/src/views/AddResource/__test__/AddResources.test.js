import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AddResource from '../AddResource';

afterEach(cleanup);

describe('Test AddResource', () => {

  it('matches snapshot', () => {
    const tree = renderer.create(<AddResource />).toJSON()
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddResource />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

});