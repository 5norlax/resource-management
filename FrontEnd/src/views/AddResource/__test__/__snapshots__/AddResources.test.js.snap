// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddResource matches snapshot 1`] = `
<div>
  <div>
    <hr />
    <h1
      className="text-center"
      data-testid="header-name"
    >
      Add Resource Form
    </h1>
    <hr />
  </div>
  <div
    className="ResourceForm container"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div>
        <h3>
          About
        </h3>
        <hr />
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="FirstName"
            >
              First Name
            </label>
            <input
              autoFocus={true}
              className="form-control"
              id="FirstName"
              name="FirstName"
              onChange={[Function]}
              placeholder="First Name"
              required={true}
              spellCheck="false"
              type="text"
              value=""
            />
          </div>
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="LastName"
            >
              Last Name
            </label>
            <input
              className="form-control"
              id="LastName"
              name="LastName"
              onChange={[Function]}
              placeholder="Last Name"
              required={true}
              spellCheck="false"
              type="text"
              value=""
            />
          </div>
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="Role"
            >
              Role
            </label>
            <select
              className="form-control"
              id="Role"
              name="Role"
              onChange={[Function]}
              placeholder="Title/Role"
              required={true}
              value=""
            >
              <option
                disabled={true}
                hidden={true}
                selected={true}
                value=""
              >
                Please Select
              </option>
              <option>
                Business Analyst
              </option>
              <option>
                Data Analyst
              </option>
              <option>
                DevOps Engineer
              </option>
              <option>
                Jr. Software Developer
              </option>
              <option>
                Manager
              </option>
              <option>
                Software Engineer
              </option>
              <option>
                Solutions Architect
              </option>
              <option>
                Sr. Technical Lead
              </option>
              <option>
                UI/UX Developer
              </option>
            </select>
          </div>
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="Email"
            >
              Email
            </label>
            <input
              className="form-control"
              id="Email"
              name="Email"
              onChange={[Function]}
              placeholder="Email Address"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="Phone"
            >
              Phone
            </label>
            <input
              className="form-control"
              id="Phone"
              name="Phone"
              onChange={[Function]}
              pattern="^[0-9]*$"
              placeholder="Phone Number (##########)"
              required={true}
              type="tel"
              value=""
            />
          </div>
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="LinkedIn"
            >
              LinkedIn
            </label>
            <input
              className="form-control"
              id="LinkedIn"
              name="LinkedIn"
              onChange={[Function]}
              placeholder="https://www.linkedin.com/in/yourlink (Optional)"
              type="url"
              value=""
            />
          </div>
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="GitHub"
            >
              GitHub
            </label>
            <input
              className="form-control"
              id="GitHub"
              name="GitHub"
              onChange={[Function]}
              placeholder="https://www.github.com/youraccount (Optional)"
              type="url"
              value=""
            />
          </div>
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="PersonalSite"
            >
              Personal Website
            </label>
            <input
              className="form-control"
              id="PersonalSite"
              name="PersonalSite"
              onChange={[Function]}
              placeholder="https://www.yoursiteurl.com (Optional)"
              type="url"
              value=""
            />
          </div>
        </div>
        <hr />
      </div>
      <div>
        <h3>
          Resume
        </h3>
        <hr />
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          >
            <div
              className="custom-file"
            >
              <input
                accept="application/pdf"
                className="custom-file-input"
                id="ResumeUpload"
                name="ResumeUpload"
                onChange={[Function]}
                type="file"
                value=""
              />
              <label
                className="custom-file-label"
                htmlFor="ResumeUpload"
              >
                Choose file
              </label>
            </div>
            <small
              className="form-text text-muted"
            >
              Please upload your most up to date resume in PDF format.
            </small>
          </div>
          <div
            className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
          />
        </div>
        <hr />
      </div>
      <div>
        <h3>
          Summary
        </h3>
        <hr />
      </div>
      <div>
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="SummaryText"
          >
            Summary
          </label>
          <textarea
            className="form-control"
            id="SummaryText"
            name="SummaryText"
            onChange={[Function]}
            placeholder="Summary (Optional)"
            value=""
          />
        </div>
        <hr />
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-md-9"
          >
            <h3
              data-testid="dynamic-header-name"
            >
              Education
            </h3>
          </div>
          <div
            className="col-md-3"
          >
            <button
              aria-label={null}
              className="section-header shadow-none btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-plus-circle fa-lg"
                style={
                  Object {
                    "color": "inherit",
                  }
                }
              />
                Add New 
              Education
            </button>
          </div>
        </div>
        <hr />
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-md-9"
          >
            <h3
              data-testid="dynamic-header-name"
            >
              Certifications
            </h3>
          </div>
          <div
            className="col-md-3"
          >
            <button
              aria-label={null}
              className="section-header shadow-none btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-plus-circle fa-lg"
                style={
                  Object {
                    "color": "inherit",
                  }
                }
              />
                Add New 
              Certifications
            </button>
          </div>
        </div>
        <hr />
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-md-9"
          >
            <h3
              data-testid="dynamic-header-name"
            >
              Experience
            </h3>
          </div>
          <div
            className="col-md-3"
          >
            <button
              aria-label={null}
              className="section-header shadow-none btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-plus-circle fa-lg"
                style={
                  Object {
                    "color": "inherit",
                  }
                }
              />
                Add New 
              Experience
            </button>
          </div>
        </div>
        <hr />
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-md-9"
          >
            <h3
              data-testid="dynamic-header-name"
            >
              Projects
            </h3>
          </div>
          <div
            className="col-md-3"
          >
            <button
              aria-label={null}
              className="section-header shadow-none btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-plus-circle fa-lg"
                style={
                  Object {
                    "color": "inherit",
                  }
                }
              />
                Add New 
              Projects
            </button>
          </div>
        </div>
        <hr />
      </div>
      <div>
        <h3>
          Skills
        </h3>
        <hr />
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12 form-group"
          >
            <label
              className=""
              htmlFor="skillsSelect"
            >
              Skills
            </label>
            <div
              className=" css-2b097c-container"
              id="select-search"
              onKeyDown={[Function]}
            >
              <div
                className=" css-yk16xz-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className=" css-g1d714-ValueContainer"
                >
                  <div
                    className=" css-1wa3eu0-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    className="css-b8ldur-Input"
                  >
                    <div
                      className=""
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="list"
                        autoCapitalize="none"
                        autoComplete="off"
                        autoCorrect="off"
                        disabled={false}
                        id="react-select-2-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        spellCheck="false"
                        style={
                          Object {
                            "background": 0,
                            "border": 0,
                            "boxSizing": "content-box",
                            "color": "inherit",
                            "fontSize": "inherit",
                            "label": "input",
                            "opacity": 1,
                            "outline": 0,
                            "padding": 0,
                            "width": "1px",
                          }
                        }
                        tabIndex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style={
                          Object {
                            "height": 0,
                            "left": 0,
                            "overflow": "scroll",
                            "position": "absolute",
                            "top": 0,
                            "visibility": "hidden",
                            "whiteSpace": "pre",
                          }
                        }
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    className=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className=" css-tlfecz-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-6q0nyr-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div>
                <input
                  name="skillsSelect"
                  type="hidden"
                />
              </div>
            </div>
          </div>
        </div>
        <hr />
      </div>
      <button
        aria-label={null}
        className="blue-button col-md-12 btn btn-primary"
        onClick={[Function]}
        type="submit"
      >
        Submit Form
      </button>
    </form>
  </div>
</div>
`;
